dapps/marketplace/src/pages/App.js

```diff
@@ -92,13 +92,16 @@ class App extends Component {
     applyConfiguration(creatorConfig)
 
     const isMobile = this.props.isMobile
-    // hide navigation bar on growth welcome screen and show it
-    // in onboarding variation of that screen
+
+    const navbarDarkMode =
+      !this.props.location.pathname.match(/^\/welcome\/onboard.*$/g) &&
+      this.props.location.pathname.match(/^\/welcome.*$/g)
 
     // TODO: Too many regex here, probably it's better to optimize this sooner or later
     const hideNavbar =
       (!this.props.location.pathname.match(/^\/welcome\/onboard.*$/gi) &&
-        this.props.location.pathname.match(/^\/welcome.*$/gi)) ||
+        this.props.location.pathname.match(/^\/welcome.*$/gi) &&
```

Comment:
We don't need the first regex. The second one will capture everything starting with `/welcome`. 

Also can we reuse the results of the previous regex match (`navbarDarkMode`)?

May be we can just use `.startsWith('welcome')` instead of regex here.

Reply:
Sorry just noted the `not` operator in the first regex. In that case, We can simplify it like `/^\/welcome\/?(?!(onboard\/)).*/gi`

Reply:
Less regexes, and simplifies the code. Thanks!